libethash-cuda/ethash_cuda_miner.cpp

```diff
@@ -204,82 +213,109 @@ bool ethash_cuda_miner::init(ethash_light_t _light, uint8_t const* _lightData, u
 {
 	try
 	{
-		int device_count = getNumDevices();
+		unsigned device_count = getNumDevices();
 
 		if (device_count == 0)
 			return false;
 
 		// use selected device
-		m_device_num = std::min<int>((int)_deviceId, device_count - 1);
-
+		unsigned device_num = _deviceId < device_count -1 ? _deviceId : device_count - 1;
```

Comment:
need to use class variable ``m_device_num``, happy to change it to unsigned, but hwmon needs the device_num later to fetch stats for the correct device.  See line 404 in this file.

Reply:
Yep...
Changing m_device_num to uint32, and all references to device_num in ethash_cuda_miner::init to m_device_num does the trick.

Reply:
Oh yeah I'm sorry... fixed it now

Reply:
@subtly is this fine now?